### ISSUE TITLE
get_replay returns a corrupt replay

### ISSUE THREAD

**Issue by osuApi** [ppy/osu-api#140]
I have tried grabbing Cookiezi's replay from xi - FREEDOM DiVE and I got a corrupt replay file back - I want to use those for analysis and the returned file is not compatible with my osu! client (the game cannot read the replay file) nor with APIs such as osuElements.

I have tried performing a base64 decode on the output and it didn't help either.

**Comment by omkelderman** [comment]
note that the response from `get_replay` is **NOT** an osr-file. See also my comments on #114

**Comment by osuApi** [comment]
I know how .osr files work. What I'm referencing to, is that the response is corrupt and even APIs like the one mentioned cannot work with it, because it's broken.

How could I get a proper replay file then?

**Comment by omkelderman** [comment]
then how is it that, when running that tool of mine, i just successfully created this file: https://omk.pics/1/Tacdf.osr

**Comment by Jeremiidesu** [comment]
Just in case, the string returned by the API contains the replay **encoded in base64**. It's not the raw replay file

**Comment by omkelderman** [comment]
thats litterally what i said in my first comment in this thread ^^

**Comment by Jeremiidesu** [comment]
Yeah true xD; though, base64 decoding the whole response is also a wrong approach @osuApi as the return of the method is shaped that way: 
```
{
    "content":"our base64 encoded replay",
    "encoding":"base64"
}
```
while it's not raw replay file, it's not raw base64-encoded replay file.
you have to decode `our base64 encoded replay` here and not the whole returned string

**Comment by omkelderman** [comment]
no shit

**Comment by Jeremiidesu** [comment]
Just in case @osuApi fucked up reading the wiki, i rewrote it here

**Comment by omkelderman** [comment]
i edited the wiki because i can

**Comment by Jeremiidesu** [comment]
Yeah i think that was needed, even I didn't get that fact. the wiki didn't explicitly tell that it didn't return a replay file but the data part of it until you edited it

**Comment by omkelderman** [comment]
well, tbf, it never said it was gonna return a replay-file, but *replay data* which is technically correct ^^

**Comment by Jeremiidesu** [comment]
People (like @osuApi) could have thought that "replay data" was the replay file but wasn't, so now it's clearer :p
